### PR TITLE
Organize Helpers and Project Structure 

### LINKED ISSUE #291: Move connectivity construction functions to separate location
Move connectivity construction functions such as `Grid._build_edge_node_connectivity()` into a separate location, like `connectivity.py`

### PR DESCRIPTION
**Pull request by philipc2**
Closes #291, #247, #101, #86
## Overview

* Project structure has been refactored, splitting the `Grid` into a separate directory with specific modules that contain Grid-Only functionality.
* Helper functions have been organized out of `utils.helpers` into specific modules
* General refactoring (cleaning up imports, organizing functions in terms of Pubic vs Private in each file, etc)
## Original Project Structure
```
.
├── ...
├── core                    
│   ├── api.py               # API
│   └── dataarray.py         # Dataarray Object
│   └── dataset.py           # Dataset Object
│   └── grid.py              # Grid Object
├── io                       
│   ├── _ugrid.py            # UGRID I/O
│   └── _exodus.py           # Exodus I/O
│   └── _mpas.py             # MPAS I/O
│   └── _scrip.py            # Scrip I/O
│   └── _shapefile.py        # Shapefile I/O
├── utils                     
│   ├── constants.py             # Top-Level constants 
│   └── get_quadratureDG.py      # Functions working with connectivity variables
│   └── helpers.py               # Assorted helper functions from all modules
```

## New Project Structure
```
.
├── ...
├── core                     
│   ├── api.py               # API
│   └── dataarray.py         # Dataarray Object
│   └── dataset.py           # Dataset Object
├── grid                     
│   ├── grid.py              # Grid Object
│   └── connectivity.py      # Functions working with connectivity variables
│   └── coordinates.py       # Functions working with coordinates
│   └── area.py              # Functions working with face area
├── io                       
│   ├── _ugrid.py            # UGRID I/O
│   └── _exodus.py           # Exodus I/O
│   └── _mpas.py             # MPAS I/O
│   └── _scrip.py            # Scrip I/O
│   └── _shapefile.py        # Shapefile I/O
│   └── utils.py             # General I/O helpers
└── constants.py             # Top-Level constants 
├── utils   # General Helper Functions
```

## PR Checklist

**General**
- [x] An issue is linked created and linked
- [x] Add appropriate labels
- [x] Filled out Overview and Expected Usage (if applicable) sections

**Documentation**
- [x] Docstrings have been added to all new functions
- [x] Docstrings have updated with any function changes
- [x] Internal functions have a preceding underscore (`_`) and have been added to `docs/internal_api/index.rst`
- [x] User functions have been added to `docs/user_api/index.rst`

### CODE REVIEW

**Review by ifranda — APPROVED** [review]
This looks good as far as I can tell!

File: uxarray/utils/numba.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import uxarray.constants
+
+
+def enable_jit_cache():
```
Comment (philipc2) [inline uxarray/utils/numba.py:4]: @aaronzedwick 

I've moved these under `uxarray.utils.numba`, but they can be accessed directly under the `uxarray.utils` module. Wanted to see if you were good with this change!

**Comment by rajeeja** [comment]
API docs seem to be broken: https://uxarray--370.org.readthedocs.build/en/370/user_api/index.html

**Comment by philipc2** [comment]
> API docs seem to be broken: https://uxarray--370.org.readthedocs.build/en/370/user_api/index.html

Will fix this.

**Comment by philipc2** [comment]
@anissa111 

Would you have an idea to why the API isn't rendering?

**Comment by philipc2** [comment]
@rajeeja 

Documentation is fixed now.

**Comment by rajeeja** [comment]
where do you think the `Analysis Operators` would go? 
in core - based on if they operate on dataset or dataarray etc?

integrate currently resides in dataset.py..

**Comment by rajeeja** [comment]
good work overall, helpers.py was getting cumbersome and this organization is better than the previous one.

**Review by rajeeja — APPROVED** [review]

**Comment by philipc2** [comment]
> where do you think the `Analysis Operators` would go? in core - based on if they operate on dataset or dataarray etc?
> 
> integrate currently resides in dataset.py..

Correct, I envision that operators that interface with both the **data** and **grid** will be housed under the `core` directory, with `grid` being used for functions that only operate on the grid and it's geometry/topology.

**Pull request merged**